### PR TITLE
replace_urls copies original configuration files to tmp

### LINKED ISSUE #44: replace_urls modifies original configuration file
`replace_urls` function in utils.py is modifying the original configuration files. Fix the function to make a copy of original files in `tmp` directory and modify the copied file.

### PR DESCRIPTION
**Pull request by sum1lim**
This PR resolves #44 

`replace_urls` is a function that stores downloaded files from `https` and modifies URLs to the stored filepaths in configuration files. Previously, it modified the content of original configuration files, which is not preferred. To avoid the problem, the function is now fixed to copy the modified content to a file in `tmp`.

### CODE REVIEW

File: osprey/utils.py [hunk 1]
```diff
@@ -61,10 +61,14 @@ def replace_urls(config, outdir):
             local_file.write(r.content)
             filedata[i] = filedata[i].replace(url, local_file.name)
 
-    with open(config, "w") as write_config:
+    config_filename = config_file.split("/")[-1]
```
Comment (nikola-rados) [inline osprey/utils.py:64]: What is the anticipated `config_file` name here? What happens if it is not in the format we expect?

**Review by nikola-rados — COMMENTED** [review]
This looks good, I just have a question about the filename. Nice work!

File: osprey/utils.py (hunk 1 above)
Comment (sum1lim) [reply osprey/utils.py:64]: Original file names are maintained. For example, if `https://docker-dev03.pcic.uvic.ca/twitcher/ows/proxy/thredds/fileServer/datasets/RVIC/sample_flow_parameters.nc` is the input, the function will return `path/to/temporary/directory/sample_flow_parameters.nc`

Comment (sum1lim) [reply osprey/utils.py:64]: > What happens if it is not in the format we expect?

Could you give an example for this?

Comment (nikola-rados) [reply osprey/utils.py:64]: I'm just wondering how this code will handle a bad path or a bad url. At what point will it be caught?

Comment (sum1lim) [reply osprey/utils.py:64]: Currently, `osprey` does not handle those. It only passes the configuration files to RVIC modules and let them handle those.

Comment (nikola-rados) [reply osprey/utils.py:64]: I forgot that the `rvic` package does basically everything for us already. In that case we don't have to worry about this.

**Pull request merged**